homeassistant/components/switch/xiaomi_miio.py

```diff
@@ -232,19 +352,69 @@ def async_update(self):
             return
 
         try:
-            state = yield from self.hass.async_add_job(self._plug.status)
+            state = await self.hass.async_add_job(self._plug.status)
             _LOGGER.debug("Got new state: %s", state)
 
             self._state = state.is_on
             self._state_attrs.update({
                 ATTR_TEMPERATURE: state.temperature,
-                ATTR_LOAD_POWER: state.load_power
+                ATTR_LOAD_POWER: state.load_power,
             })
 
+            if self.supported_features & SUPPORT_SET_POWER_MODE == 1 \
+                    and state.mode:
+                self._state_attrs[ATTR_POWER_MODE] = state.mode.value
+
+            if self.supported_features & SUPPORT_SET_WIFI_LED == 1 and \
+                    state.wifi_led:
+                self._state_attrs[ATTR_WIFI_LED] = state.wifi_led
+
+            if self.supported_features & SUPPORT_SET_POWER_PRICE == 1 and \
+                    state.power_price:
+                self._state_attrs[ATTR_POWER_PRICE] = state.power_price
+
         except DeviceException as ex:
             self._state = None
             _LOGGER.error("Got exception while fetching the state: %s", ex)
 
+    async def async_set_power_mode(self, mode: str):
+        """Set the power mode."""
+        if self.supported_features & SUPPORT_SET_POWER_MODE == 0:
```

Comment:
Same as above wrt. trueness check. Maybe someone else wants to chime in what form is preferred?

Reply:
Could somebody recommend the preferred way?